include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -89,12 +90,25 @@ class pool_memory_resource final
         return upstream_mr;
       }()}
   {
+    RMM_EXPECTS(
```

Comment:
Docs should be updated to reflect that these args need to be a multiple of `pool_memory_resource::allocation_alignment`.

Reply:
Added @throws docs.